src/lib/classes/access-control-list.js

```diff
@@ -0,0 +1,333 @@
+      predicates = [...originalPredicates, publicQuad];
+    }
+
+    const quadList = modes.reduce(
+      (array, mode) => [
+        ...array,
+        this.createQuad(subject, `${acl}mode`, namedNode(`${acl}${mode}`))
+      ],
+      predicates
+    );
+
+    return quadList;
+  };
+
+  /**
+   * @function createPermissionsTurtle Creates a turtle looking string with specific permissions
+   * @param {Array<Permissions> | null} permissions Array of permissions to be added to the turtle string
+   * @return {String } A Turtle looking string with all of the necessary permissions
+   */
+  createPermissionsTurtle = (permissions: Array<Permissions>) => {
+    const { DataFactory } = N3;
+    const prefixes = { ...ACL_PREFIXES, '': `${this.aclUri}#`, me: this.owner };
+    const { namedNode, quad } = DataFactory;
+    const writer = new N3.Writer({ prefixes });
+    const quadPermissions = permissions.map(({ modes, agents }) =>
+      this.createQuadList(modes, agents)
+    );
+    const quads = quadPermissions
+      .map(quadItem => {
+        const itemQuads = quadItem.map(({ subject, predicate, object }) =>
+          quad(namedNode(subject), namedNode(predicate), object)
+        );
+        return itemQuads;
+      })
+      .reduce((array, item) => [...array, ...item], []);
+
+    quads.forEach(quad => writer.addQuad(quad));
+    let turtle;
+    writer.end((error, result) => {
+      if (!error) turtle = result;
+    });
+    return turtle;
+  };
+
+  /**
+   * @function createACLFile Creates a file or a container with it specfic set of acls. Assigns READ, WRITE and CONTROL permissions to the owner by default
+   * @param {Array<Permissions> | null} permissions Array of permissions to be added in the acl file
+   */
+  createACLFile = async (permissions = null) => {
+    await this.createSolidFile(this.documentUri);
+    if (permissions) {
+      const permissionList = [
+        { agents: this.owner, modes: [PERMISSIONS.READ, PERMISSIONS.WRITE, PERMISSIONS.CONTROL] },
+        ...permissions
+      ];
+      const body = this.createPermissionsTurtle(permissionList);
+      await this.createSolidFile(this.aclUri, { body });
+    }
+  };
+
+  /**
+   * @function void Helper function to create a file
+   * @param {String} url Url where the solid file has to be created
+   * @param {Object} options Options to add as part of the native fetch options object
+   */
+  createSolidFile = async (url: String, options: Object = {}) =>
+    solid.fetch(url, {
+      method: 'PUT',
+      headers: {
+        'Content-Type': 'text/turtle'
+      },
+      ...options
+    });
+
+  /**
+   * @function getParentACL Recursively tries to retrieve the acl file from the parent containers if there is not a direct one
+   * @param {String} url Url of the parent container
+   * @returns {Object} Parent acl fetched file
+   */
+  getParentACL = async (url: String) => {
```

Comment:
It looks like there might be a codepath in this function where nothing gets returned?

Reply:
I Added return null by default